src/lib.rs

```diff
@@ -102,48 +104,49 @@ impl MemoryListener {
     /// use memory_socket::MemoryListener;
     ///
     /// # fn main () -> ::std::io::Result<()> {
-    /// let listener = MemoryListener::bind(16)?;
+    /// let listener = MemoryListener::bind("192.51.100.2:1337")?;
     /// # Ok(())}
     /// ```
-    pub fn bind(port: u16) -> Result<Self> {
+    pub fn bind<A: ToSocketAddrs>(addresses: A) -> Result<Self> {
```

Comment:
A completely separate problem is that there are three (at least) different `ToSocketAddrs`, and none of them is interchangeable. `std::net::ToSocketAddrs` is synchronous; `async_std::net::ToSocketAddrs` is async because it has to do name resolution, and `tokio::net::ToSocketAddrs` is similarly async.

The most obvious solution would be to make bind and connect accept a single `SocketAddr` instead of a ToSocketAddrs. That breaks the drop-in replacement dream, though, but I've kind of given up on that anyway...

Reply:
right. in my mind, memsocket should not be doing any IO. i think it makes sense to just take a `SocketAddr` here like [`mio::net::TcpListener::bind`](https://docs.rs/mio/0.7.0/mio/net/struct.TcpListener.html).

if we really wanted to take this to it's natural conclusion, we could setup a memory-dns that allows you to register a domain name "example.com" with its ip addrs. then we include a local `memory-socket::ToSocketAddrs` trait that uses this in-memory dns for domain lookups.

of course, this seems like far more effort and complexity than it's worth. so i'm happy with just binding a `SocketAddr` : )